classes/helper.php

```diff
@@ -0,0 +1,127 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_hvp;
+
+/**
+ * Helper class for plugin.
+ *
+ * @package    mod_hvp
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class helper {
+
+    /**
+     * Get an installed library.
+     *
+     * @param array $options Array containing either id or a title and version number.
+     * @return \stdClass Record from hvp_libraries.
+     */
+    public static function get_library(array $options): \stdClass {
+        global $DB;
+        if (!empty($options['library-id'])) {
+            $libraries = $DB->get_records('hvp_libraries', ['id' => $options['library-id']]);
+        } else if (!empty($options['library-title']) && !empty($options['library-version'])) {
+            $versionparts = explode('.', $options['library-version']);
+            if (empty($versionparts) || count($versionparts) !== 3) {
+                throw new \moodle_exception('error:invalidversionformat', 'hvp');
+            }
+            $libraries = $DB->get_records('hvp_libraries', [
+                'title' => $options['library-title'],
+                'major_version' => $versionparts[0],
+                'minor_version' => $versionparts[1],
+                'patch_version' => $versionparts[2],
+            ]);
+        } else if (!empty($options['library-title'])
+                && !empty($options['major-version'])
+                && !empty($options['minor-version'])
+                && !empty($options['patch-version'])) {
+            $libraries = $DB->get_records('hvp_libraries', [
+                'title' => $options['library-title'],
+                'major_version' => $options['major-version'],
+                'minor_version' => $options['minor-version'],
+                'patch_version' => $options['patch-version'],
+            ]);
+        }
+
+        if (empty($libraries)) {
+            throw new \moodle_exception('error:nolibraryexists', 'hvp');
+        }
+
+        if (count($libraries) !== 1) {
+            throw new \moodle_exception('error:multiplelibrariesfound', 'hvp');
```

Comment:
I dont really think this should be an error condition. Infact, this is ideal for deleting any old versions etc

Reply:
Due to how delicate the versions are with various dependencies, all library versions should not be deleted at the same time. Also, there is no uniqueness restriction with the title so technically 2 different libraries could have the same title and version.

Reply:
No worries. I agree, if its potentially unsafe, lets just make it single.